server/configuration-validation.js

```diff
@@ -1,127 +1,195 @@
 const { isObject } = require('./utils')
 
-/**
- * Validates the plugin configuration provided in `plugins/config.js` of the users plugin configuration.
- * Modifies the value of config on place.
- *
- * @param  {object} config - configurations
- */
-function validateConfiguration(config) {
-  if (config === undefined) {
-    return
-  }
+function CollectionConfig(collectionName, configuration) {
+  const log = strapi.log // has to be inside a scope
+  const {
+    indexName,
+    transformEntry,
+    filterEntry,
+    settings,
+    populateEntryRule,
+    ...excedent
+  } = configuration
```

Comment:
Wouldn't it be better if you go for `collection` instead of `configuration`?

Reply:
Since its the configuration of the collection. If you want I can call it `collectionConfig`, but since we are already in `CollectionConfig` it seems clear for me that `conifguration` is the specific configuration of that collection

Reply:
Ok, it was a bit confusing for me as what you send to this function is:
```js
CollectionConfig(
  collection,
  collections[collection]
)
```
which to me was a collection (including the config in it).

But it does not matter, I don't mind if we don't rename it on one side.